elixir/apps/web/lib/web/live/settings/dns.ex

```diff
@@ -1,7 +1,113 @@
 defmodule Web.Settings.DNS do
   use Web, :live_view
 
+  alias Domain.Config
+
+  defp remove_errors(changeset, field, message) do
+    errors =
+      changeset.errors
+      |> Enum.filter(fn err ->
+        case err do
+          {^field, {msg, _}} ->
+            if msg == message, do: false, else: true
+
+          {_, _} ->
+            true
+        end
+      end)
+
+    %{changeset | errors: errors}
+  end
+
+  defp filter_errors(changeset) do
+    filtered_cs =
+      changeset
+      |> remove_errors(:clients_upstream_dns, "address can't be blank")
```

Comment:
if we did not add `required: true` to `cast_embed/3` then we should never have this error.

Reply:
`required: true` is not added to the `cast_embed/3` call, but this error does seem to show up in the changeset when one of `clients_upstream_dns` addresses is blank.

`validate_required` is called in the ` clients_upstream_dns_changeset/2` function though.  I made the assumption that calling `validate_required` in the changeset for the embedded schema somehow got propagated backup to the top level changeset.  Should this not be happening?